panic: close of closed channel

### Bug report

__System info:__ [Include InfluxDB version, operating system name, and other relevant details]
version 1.4.2, branch 1.4, commit 6d2685d1738277a1c2672fc58df7994627769be6

__Steps to reproduce:__
 Not sure. The database server has been stable for weeks.

__Expected behavior:__ Server doesn't panic and crash.

__Actual behavior:__ Server panic'ed and crashed.

__Additional info:__
```
panic: close of closed channel
goroutine 28674303 [running]:
github.com/influxdata/influxdb/query.(*QueryTask).kill(0xc43b0a3180, 0xc420216630, 0x240a8)
/go/src/github.com/influxdata/influxdb/query/query_executor.go:523 +0x8e
github.com/influxdata/influxdb/query.(*TaskManager).KillQuery(0xc420209620, 0x240a8, 0x10d14a0, 0xc42003b010)
/go/src/github.com/influxdata/influxdb/query/task_manager.go:206 +0x90
github.com/influxdata/influxdb/query.(*TaskManager).waitForQuery(0xc420209620, 0x240a8, 0xc422182060, 0xc4bdad9680, 0xc422182120)
/go/src/github.com/influxdata/influxdb/query/task_manager.go:274 +0x1a7
created by github.com/influxdata/influxdb/query.(*TaskManager).AttachQuery
/go/src/github.com/influxdata/influxdb/query/task_manager.go:176 +0x335
```

Immediately preceding those log entries were these (I'm not sure if they're relevant, since they appear to be successful events):
```
[I] 2017-12-10T02:45:50Z executing continuous query runner_cpu_daily (2017-12-10 02:45:40 +0000 UTC to 2017-12-10 02:45:50 +0000 UTC) service=continuous_querier
[I] 2017-12-10T02:45:50Z finished continuous query runner_cpu_daily, 1 points(s) written (2017-12-10 02:45:40 +0000 UTC to 2017-12-10 02:45:50 +0000 UTC) in 4.59884ms service=continuous_querier
[I] 2017-12-10T02:45:50Z finished continuous query runner_cpu_daily, 1 points(s) written (2017-12-10 02:45:40 +0000 UTC to 2017-12-10 02:45:50 +0000 UTC) in 4.59884ms service=continuous_querier
```

Otherwise there is normal incoming traffic from our own clients and telegraf.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -28,6 +28,7 @@
 - [#9230](https://github.com/influxdata/influxdb/pull/9230): Remove extraneous newlines from the log.
 - [#9226](https://github.com/influxdata/influxdb/issues/9226): Allow lone boolean literals in a condition expression.
 - [#9235](https://github.com/influxdata/influxdb/pull/9235): Improve performance when writes exceed `max-values-per-tag` or `max-series`.
+- [#9216](https://github.com/influxdata/influxdb/issues/9216): Prevent a panic when a query simultaneously finishes and is killed at the same time.
 
 ## v1.4.3 [unreleased]
 

--- a/query/query_executor.go
+++ b/query/query_executor.go
@@ -508,6 +508,8 @@ func (q *QueryTask) monitor(fn QueryMonitorFunc) {
 func (q *QueryTask) close() {
 	q.mu.Lock()
 	if q.status != KilledTask {
+		// Set the status to killed to prevent closing the channel twice.
+		q.status = KilledTask
 		close(q.closing)
 	}
 	q.mu.Unlock()

